src/main/cljc/jarl/builtins/encoding.cljc

```diff
@@ -0,0 +1,137 @@
+(ns jarl.builtins.encoding
+  (:require [jarl.encoding.base64 :as base64]
+            [jarl.encoding.json :as json]
+            [jarl.encoding.hex :as hex]
+            [jarl.exceptions :as errors]
+            [jarl.utils :refer [url-decode url-encode]]
+            [clojure.string :as str]
+            #?(:clj [clj-yaml.core :as yaml]))
+  #?(:clj (:import (java.io EOFException))))
+
+(defn builtin-base64-encode
+  [{[^String s] :args}]
+  (base64/encode s))
+
+(defn builtin-base64-decode
+  [{[^String s] :args}]
+  (base64/decode s))
+
+(defn builtin-base64-url-encode
+  [{[^String s] :args}]
+  (base64/url-encode s))
+
+(defn builtin-base64-url-encode-no-pad
+  [{[^String s] :args}]
+  (base64/url-encode-no-pad s))
+
+(defn builtin-base64-url-decode
+  [{[^String s] :args}]
+  (base64/url-decode s))
+
+(defn builtin-url-query-encode
+  [{[^String s] :args}]
+  (url-encode s))
+
+(defn builtin-url-query-encode-object
+  [{[o] :args}]
+  (let [kv-str (fn [[k vals]]
+                 (let [vals (if (string? vals) [vals] vals)]
+                   (str/join "&" (for [v vals] (str (url-encode k) "=" (url-encode v))))))]
+    (str/join "&" (map kv-str o))))
+
+(defn builtin-url-query-decode
+  [{[^String s] :args}]
+  (url-decode s))
+
+(defn builtin-url-query-decode-object
+  [{[^String s] :args}]
+  (let [assoc-param (fn [map key val] (assoc map key (conj (get map key []) val)))]
+    (reduce
+      (fn [param-map encoded-param]
+        (if-let [[_ key val] (re-matches #"([^=]+)=?(.*)" encoded-param)]
+          (assoc-param param-map (url-decode key) (url-decode (or val "")))
+          param-map))
+      {}
+      (str/split s #"&"))))
+
+(defn builtin-json-marshal
+  [{[x] :args}]
+  #?(:clj
```

Comment:
Yeah, that's pretty much what we had to start with, but I had to rewrite the exception helpers to invoke functions rather than forms (i.e macros) because of the ClojureScript limitation of macros being a compile-time thing, and catching/mapping exceptions a runtime concern. It might be doable, but I spent way more time on that than I'd like to admit already, so will leave it as it is for the time being 😬